packages/node/src/integrations/contextlines.ts

```diff
@@ -27,62 +21,210 @@ interface ContextLinesOptions {
   frameContextLines?: number;
 }
 
-/** Exported only for tests, as a type-safe variant. */
-export const _contextLinesIntegration = ((options: ContextLinesOptions = {}) => {
-  const contextLines = options.frameContextLines !== undefined ? options.frameContextLines : DEFAULT_LINES_OF_CONTEXT;
+/**
+ * Exported for testing purposes.
+ */
+export function resetFileContentCache(): void {
+  LRU_FILE_CONTENTS_CACHE.clear();
+}
 
-  return {
-    name: INTEGRATION_NAME,
-    processEvent(event) {
-      return addSourceContext(event, contextLines);
-    },
-  };
-}) satisfies IntegrationFn;
+/**
+ * Get or init map value
+ */
+function emplace<T extends LRUMap<K, V>, K extends string, V>(map: T, key: K, contents: V): V {
+  const value = map.get(key);
+
+  if (value === undefined) {
+    map.set(key, contents);
+    return contents;
+  }
+
+  return value;
+}
 
 /**
- * Capture the lines before and after the frame's context.
+ * Determines if context lines should be skipped for a file.
+ * - .min.(mjs|cjs|js) files are and not useful since they dont point to the original source
+ * - node: prefixed modules are part of the runtime and cannot be resolved to a file
+ * - data: skip json, wasm and inline js https://nodejs.org/api/esm.html#data-imports
  */
-export const contextLinesIntegration = defineIntegration(_contextLinesIntegration);
+const SKIP_CONTEXTLINES_REGEXP = /(^(node|data):)|(\.min\.(mjs|cjs|js$))/;
+function shouldSkipContextLinesForFile(path: string): boolean {
+  return SKIP_CONTEXTLINES_REGEXP.test(path);
```

Comment:
I'm honestly not sure since startsWith and endsWith wont accept regexp. That said, this wont be performed often and we can probably sort the statements so that we return on most common cases first, which I would suspect is what will make this faster. I'll change it